sdk/azidentity/chained_token_credential.go

```diff
@@ -62,36 +64,40 @@ func (c *ChainedTokenCredential) GetToken(ctx context.Context, opts policy.Token
 		token, err = cred.GetToken(ctx, opts)
 		var credErr credentialUnavailableError
 		if errors.As(err, &credErr) {
-			errList = append(errList, credErr)
+			errList = append(errList, err.(credentialUnavailableError))
 		} else if err != nil {
 			var authFailed AuthenticationFailedError
 			if errors.As(err, &authFailed) {
-				err = fmt.Errorf("Authentication failed:\n%s\n%s"+createChainedErrorMessage(errList), err)
+				err = fmt.Errorf("ERROR in GetToken() call for %s:%s\n\t%s", c.name, createChainedErrorMessage(errList), err)
 				authErr := newAuthenticationFailedError(err, authFailed.RawResponse)
 				return nil, authErr
 			}
 			return nil, err
 		} else {
-			logGetTokenSuccess(c, opts)
+			log.Writef(EventAuthentication, "Azure Identity => GetToken() result for %s %s:", c.name, extractCredentialName(cred))
```

Comment:
This logs a message like `Azure Identity => GetToken() result for DefaultAzureCredential AzureCLICredential:`. How about
```suggestion
			log.Writef(EventAuthentication, "Azure Identity => %s authenticated with %s", c.name, extractCredentialName(cred))
```
which would give a message like `Azure Identity => DefaultAzureCredential authenticated with AzureCLICredential`?

Reply:
I love it! thanks!